### PR TITLE
[EOSF-824] Allow quickfiles to create guids, and reroute based on fid without no…

### PR DESCRIPTION
**Pull request by hmoco**
## Ticket:
https://openscience.atlassian.net/browse/EOSF-824

## Purpose
For viewing quickfiles: Add new route, like /quickfiles/<path:path>/ that proxies to /project/<quickfiles-guid>/files/osfstorage/<path:path>/
For generating share links: Bypass admin permission check when creating GUIDs for quickfiles when using ?create_guid=1

### CODE REVIEW

File: api/files/views.py [hunk 1]
```diff
@@ -336,13 +336,12 @@ def get_node(self):
...
     def get_object(self):
         user = utils.get_user_auth(self.request).user
-
-        if (self.request.GET.get('create_guid', False) and
-                self.get_node().has_permission(user, 'admin') and
-                utils.has_admin_scope(self.request)):
-            self.get_file(check_permissions=True).get_guid(create=True)
-
-        return self.get_file()
+        file = self.get_file()
```
Comment (alexschiller) [inline api/files/views.py]: very, very minor driveby: `file` is a `type` in python (which is why it gets highlighted funny like that). Better to use a different name (although it is used all over in the OSF so not a big deal if you aren't making other changes).

**Comment by sloria** [comment]
@hmoco Ticket for views test: https://openscience.atlassian.net/browse/EOSF-835
